### PR TITLE
Eval loop adjustments.

### PR DESCRIPTION
**Pull request by Sid-Data-Universe**
1) Adjust temperature to concentrate incentive on top model further.
2) Adjust from 20 baseline + 30 new models to 6 baseline + 14 new models to reduce overall loop time.
3) Prioritize checking for model updates to models with incentive on the chain.

### CODE REVIEW

File: neurons/validator.py [hunk 1]
```diff
@@ -370,26 +376,57 @@ def update_models(self):
...
+                with self.metagraph_lock:
+                    incentives = self.metagraph.I
+
+                for uid, incentive in enumerate(incentives):
+                    # Use .item() to get the number value since this is a tensor.
+                    if (
+                        incentive.item()
+                        >= constants.update_priority_incentive_threshold
+                    ):
+                        # Confirm that we haven't checked it in the last 30 minutes in this path.
```
Comment (RusticLuftig) [inline neurons/validator.py:393]: Should we make this 20 minutes to align with the cadence the chain metadata can be updated?

File: neurons/validator.py [hunk 2]
```diff
@@ -370,26 +376,57 @@ def update_models(self):
...
-                if time_diff and time_diff < dt.timedelta(minutes=5):
-                    # If we have seen it within 5 minutes then sleep until it has been at least 5 minutes.
-                    time_to_sleep = (
-                        dt.timedelta(minutes=5) - time_diff
-                    ).total_seconds()
-                    bt.logging.trace(
-                        f"Update loop has already processed all UIDs in the last 5 minutes. Sleeping {time_to_sleep} seconds."
+                # First check for any uids with incentives above threshold on the chain.
+                # This will catch updates to current best models and models other valis have incentivized faster.
+                incentives = None
```
Comment (RusticLuftig) [inline neurons/validator.py:383]: nit: not needed. a "with" statement doesn't create a new variable scope

File: neurons/validator.py [hunk 3]
```diff
@@ -370,26 +376,57 @@ def update_models(self):
...
+                        ).total_seconds()
+                        bt.logging.trace(
+                            f"Update loop has already processed all UIDs in the last 5 minutes. Sleeping {time_to_sleep} seconds."
+                        )
+                        time.sleep(time_to_sleep)
+
+                    uid_last_checked_sequential[next_uid] = dt.datetime.now()
 
                 # Get their hotkey from the metagraph.
                 hotkey = "NoHotkey"
```
Comment (RusticLuftig) [inline neurons/validator.py:432]: existing nit: not needed

File: temp-check.ipynb
Comment (RusticLuftig) [inline temp-check.ipynb:1]: Remove?

**Review by RusticLuftig — APPROVED** [review]

File: neurons/validator.py (hunk 1 above)
Comment (Sid-Data-Universe) [reply neurons/validator.py]: Done. And below.

File: temp-check.ipynb
Comment (Sid-Data-Universe) [reply temp-check.ipynb:1]: Had added it intentionally, but in retrospect we will never plan to update the temperature again so it isn't really worth keeping around.

File: neurons/validator.py (hunk 3 above)
Comment (Sid-Data-Universe) [reply neurons/validator.py]: Done.

**Review by RusticLuftig — APPROVED** [review]

**Review by RusticLuftig — APPROVED** [review]

**Pull request merged**